OrbitGgp/Instance.cpp

```diff
@@ -10,59 +10,65 @@
 #include <QJsonObject>
 #include <QJsonValue>
 
+#include "OrbitGgp/Error.h"
+
 namespace OrbitGgp {
 
 namespace {
 
-Instance GetInstanceFromJson(const QJsonObject& obj) {
-  Instance inst{};
-
-  if (const QJsonValue val = obj.value("displayName"); !val.isUndefined()) {
-    inst.display_name = val.toString();
-  }
-
-  if (const QJsonValue val = obj.value("id"); !val.isUndefined()) {
-    inst.id = val.toString();
-  }
-
-  if (const QJsonValue val = obj.value("ipAddress"); !val.isUndefined()) {
-    inst.ip_address = val.toString();
+outcome::result<Instance> GetInstanceFromJson(const QJsonObject& obj) {
+  const QJsonValue display_name_val = obj.value("displayName");
```

Comment:
Maybe a lambda here?
```cpp
const auto process = [](const QJsonValue& val) -> outcome::result<QString> {
  if (!val.isString()) {
    return Error::kUnableToParse;
  } else {
    return val.toString();
  }
};

OUTCOME_TRY(display_name, process(obj.value("displayName")));
OUTCOME_TRY(id, process(obj.value("id")));
...
```

Reply:
yes, makes it nicer; done here and in SshInfo.cpp